src/Context/product-context.js

```diff
@@ -0,0 +1,637 @@
+        genre: "tech",
+        rating : 4   
+    },
+    {
+        id : 21,
+        bookname : "You Don't Know JS",
+        author   : "Kyle Simpson",
+        originalprice : 800,
+        discountedprice : 600,
+        discountpercent : 25,
+        imgsrc : You_Dont_Know_JS,
+        imgalt : 'Book-Dan Brown - You Dont Know JS',
+        badgetext : 'Tech',
+        outOfStock: false,
+        fastDeliveryAvailable : true,
+        genre: "tech",
+        rating : 4    
+    }
+]
+
+function productsOrderFunc(state,action)
+{
+    switch(action.type)
+    {
+        case "PRICE_HIGH_TO_LOW" : 
+            {
+                for(let i=0; i<state.length-1; i++)
+                {
+                    for(let j=0; j<state.length-1-i; j++)
+                    {
+                        if(state[j].discountedprice < state[j+1].discountedprice && i+1<state.length)
+                        {
+                            let swappedNum = state[j];
+                            state[j] = state[j+1];
+                            state[j+1] = swappedNum
+                        }
+                    }
+                }
+                return [...state]
+            }
+
+        case "PRICE_LOW_TO_HIGH" : 
+            {
+                for(let i=0; i<state.length-1; i++)
+                {
+                    for(let j=0; j<state.length-1-i; j++)
+                    {
+                        if(state[j].discountedprice > state[j+1].discountedprice && i+1<state.length)
+                        {
+                            let swappedNum = state[j];
+                            state[j] = state[j+1];
+                            state[j+1] = swappedNum
+                        }
+                    }
+                }
+                return [...state]
+            }
+
+        case "UPDATE_LIST_AS_PER_FILTERS" : 
+            {
+                if(action.payload.includeOutOfStock===true)
+                {
+                    //All products
+                    return (action.payload.onlyFastDeliveryProducts===false) 
+                    ? productList.filter(item=>
+                        (
+                            action.payload.minPrice<=item.discountedprice 
+                            && item.discountedprice<action.payload.maxPrice
+                            && action.payload[item.genre]
+                            && item.rating>=action.payload.minRating
+                        )
+                      )
+                    : productList.filter(item=> 
+                        (
+                            item.fastDeliveryAvailable===true
+                            && (
+                                action.payload.minPrice<=item.discountedprice 
+                                && item.discountedprice<action.payload.maxPrice
+                               ) 
+                            && action.payload[item.genre]
+                            && item.rating>=action.payload.minRating
```

Comment:
Suggestion: You can pass a diff function inside filter it would increase readability